### PR TITLE
Update components with Mantine Component Library

### LINKED ISSUE #85: FE: Look into implementing a component library
Some of the code seems like you did stuff from scratch instead of using a component library with those features built in (particularly the form and autocomplete). A component library also makes the theming/styles central, so you won't have to spend as much time building and styling things like buttons or text inputs and can also follow any iNat style guidelines. 

Definition of Done:
- Research and choose a component library (my go to is Mantine but Material UI is also good)
- Switch out your code for the components/hooks provided by the library 
- Change the theme to follow the styles/color scheme you decide on

there's some comments in here about imposing a folder structure, code refactoring 
original prototype pr: https://github.com/UMassCDS/ds4cg2024-inaturalist/pull/25#pullrequestreview-2170030686

### PR DESCRIPTION
**Pull request by oz03-hub**
## Added
- mantine component library components
- frontend is switched to vite+react, recommended by mantine library
- Application now uses Vite build tool
- Added new visuals for loading status
## Updated
- README updated to reflect new application url and instruction on running Vite
- Converted old JS components into JSX components and updated components with mantine counterparts
- frontend dockerfile to build vite before serving
- Broke up functions in App.js into separate functions in utils.js, makes it more readable
## Removed
- old css files that are now unused
- loadingbarstatus.js, now we use loading overlays and display errors with alerts

## How to test
You can still follow regular instructions in readme to start local instance, docker is not working because of backend-postgres connection, explained below.

Two important things: make sure to run `npm i` and then `npm run dev` to start the application, it will be served in `localhost:5173`

## Problems
I noticed that docker-compose removed env-file option, I was not able to run backend because it needs a specific url, I think this was done for deployment, or I just have incorrect url in my project configurations

Leaflet has a known issue: https://github.com/Leaflet/Leaflet.draw/issues/1026, it is not patched yet and causes us to create a global variable to not the throw an error

Closes #85

### CODE REVIEW

**Review by ginic — CHANGES_REQUESTED** [review]
@oz03-hub , this looks great! I tested out both using Docker and running locally and everything is working as expected for me. I really like these changes! The UI looks nicer and has better behaviour with screen resizing and I also think these changes make the code easier to follow. Really nice work! 

I noticed a couple minor things that are worth checking over. There are more details in comments, but basically it comes down to two things: 
- Are you sure we need a new package-lock.json and package.json at the root level? 
- Check the error handling between your functions in api.js and utils.js and make sure it's working in the way you expect so that it's easy to debug any errors

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
[the remark this answers is not in the document]
Comment (oz03-hub) [reply package-lock.json:1]: These files should not be here, I will remove them, I guess that's why the npm i you ran placed things in the root

File: src/frontend/src/api.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export async function generatePrediction(data) {
+  const response = await fetch(`${API_URL}/generate_prediction/`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data),
+  })
+    .then((response) => response.json())
+    .catch((error) => console.error("Error generating prediction:", error));
```
[the remark this answers is not in the document]
Comment (oz03-hub) [reply src/frontend/src/api.js:15]: I understand, I removed catch from api functions, any error handling is done in util.js

**Comment by oz03-hub** [comment]
Removed root packages (unnecessary), removed unused css, make error handling only handled by caller

File: src/frontend/src/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ginic) [reply src/frontend/src/api.js]: Thanks! The changes look good to me!

**Review by ginic — APPROVED** [review]
Nice work! This changes are excellent!

**Pull request merged**